tests/CSharpTemp/CSharpTemp.Tests.cs

```diff
@@ -344,4 +345,12 @@ public void TestNullAttributedFunctionPtr()
         }
     }
 
+    [Test]
+    public unsafe void TestMarshallingOfCharPtr()
+    {
+        var obj = new CheckMarshllingOfCharPtr();
+        var str = obj.FuncRetCharPtr;
+        Assert.AreEqual('S', str[0]);
+        Assert.AreEqual('i', str[3]);
```

Comment:
What values do you get instead? Also, please try this in native code instead: str = "String". It should be the same but it's simpler and it's actually the way C/C++ strings are used in practice.